### PR TITLE
Test to validate storage cluster status on UI

### LINKED ISSUE #6602: Validate ODF dashboard presence after OCP and ODF upgrades

### PR DESCRIPTION
**Pull request by am-agrawa**
Signed-off-by: am-agrawa <amagrawa@redhat.com>

### CODE REVIEW

File: ocs_ci/ocs/ocs_upgrade.py [hunk 1]
```diff
@@ -741,6 +747,13 @@ def run_ocs_upgrade(operation=None, *operation_args, **operation_kwargs):
             version_before_upgrade=upgrade_ocs.version_before_upgrade,
         )
 
+    # Login to OCP console and run ODF dashboard validation check
+    ocs_version = version.get_semantic_version(version="4.9", only_major_minor=True)
+    if upgrade_version >= ocs_version:
```
Comment (petr-balogh) [inline ocs_ci/ocs/ocs_upgrade.py:752]:
```suggestion
    semantic_upgrade_version = version.get_semantic_version(upgrade_version, True)
    if semantic_upgrade_version >= version.VERSION_4_9:
```

File: tests/ecosystem/upgrade/test_upgrade_ocp.py [hunk 2]
```diff
@@ -159,6 +161,14 @@ def test_upgrade_ocp(self, reduce_and_resume_cluster_load):
         cluster_ver = ocp.run_cmd("oc get clusterversions/version -o yaml")
         logger.debug(f"Cluster versions post upgrade:\n{cluster_ver}")
 
+        # Login to OCP console and run ODF dashboard validation check
+        version_post_upgrade = parse_version(ocp_upgrade_version)
+        ocs_version = version.get_semantic_version(version="4.9", only_major_minor=True)
```
Comment (petr-balogh) [inline tests/ecosystem/upgrade/test_upgrade_ocp.py:166]: no need, and you can directly use later on in the condition version.VERSION_4_9

File: tests/ecosystem/upgrade/test_upgrade_ocp.py [hunk 3]
```diff
@@ -159,6 +161,14 @@ def test_upgrade_ocp(self, reduce_and_resume_cluster_load):
...
+        if version_post_upgrade >= ocs_version:
```
Comment (petr-balogh) [inline tests/ecosystem/upgrade/test_upgrade_ocp.py:167]:
```suggestion
       
        if version_post_upgrade >= version.VERSION_4_9:
```

Comment (am-agrawa) [reply tests/ecosystem/upgrade/test_upgrade_ocp.py]: > 

I am getting this error again:

2022-11-25 16:50:10  >       if version_post_upgrade >= version.VERSION_4_9:
2022-11-25 16:50:10  E       TypeError: '>=' not supported between instances of 'LegacyVersion' and 'Version'
2022-11-25 16:50:10  
2022-11-25 16:50:10  tests/ecosystem/upgrade/test_upgrade_ocp.py:166: TypeError

File: tests/ecosystem/upgrade/test_upgrade_ocp.py [hunk 4]
```diff
@@ -159,6 +161,14 @@ def test_upgrade_ocp(self, reduce_and_resume_cluster_load):
         cluster_ver = ocp.run_cmd("oc get clusterversions/version -o yaml")
         logger.debug(f"Cluster versions post upgrade:\n{cluster_ver}")
 
+        # Login to OCP console and run ODF dashboard validation check
+        version_post_upgrade = parse_version(ocp_upgrade_version)
```
[the remark this answers is not in the document]
Comment (petr-balogh) [reply tests/ecosystem/upgrade/test_upgrade_ocp.py:165]: you are still using legacy parse_version here, you need to replace this with what I suggested above 
```suggestion
        version_post_upgrade = version.get_semantic_version(ocp_upgrade_version, True)
```

**Comment by am-agrawa** [comment]
Validation passed when OCP+ODF 4.11 is upgraded to OCP+ODF 4.12
https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/19553/testReport/

**Comment by am-agrawa** [comment]
Validation passed locally on OCP+ODF 4.12- http://pastebin.test.redhat.com/1086997

**Comment by am-agrawa** [comment]
Verified against 4.14- http://pastebin.test.redhat.com/1104133

File: tests/ui/test_validation_ui.py [hunk 5]
```diff
@@ -31,6 +31,21 @@ class TestUserInterfaceValidation(object):
 
     """
 
+    @ui
+    @tier1
+    @black_squad
```
Comment (ebenahar) [inline tests/ui/test_validation_ui.py:36]: missing polarion id marker

**Review by ebenahar — APPROVED** [review]

**Pull request merged**